runner/src/server/plugins/engine/pluginHandlers/files/prehandlers/handleUpload.ts

```diff
@@ -108,9 +102,13 @@ export async function handleUpload(
         loggerIdentifier,
         `Document upload API responded with an error ${error}`
       );
-      request.pre.errors = [
-        ...(request.pre.errors || []),
-        parsedError(fieldName, error),
+      errors.add(error)
+    }
+
+    for (let e of errors) {
+        request.pre.errors = [
+        ...(h.request.pre.errors || []),
+        parsedError(fieldName, e),
```

Comment:
I'm not sure you intend to be reassigning `request.pre.errors` for each iteration of this loop - you'll end up just with `h.request.pre.errors` and the last error in `errors`. Instead something like (untested):
```
request.pre.errors = [
        ...(h.request.pre.errors || []),
        ...errors.map(e => parsedError(fieldName, e)),
];
```